offer-manager-domain/src/main/groovy/life/qbic/business/offers/content/CreateOfferContent.groovy

```diff
@@ -0,0 +1,176 @@
+import life.qbic.datamodel.dtos.business.OfferId
+import life.qbic.datamodel.dtos.business.ProductItem
+import life.qbic.datamodel.dtos.business.services.DataStorage
+import life.qbic.datamodel.dtos.business.services.MetabolomicAnalysis
+import life.qbic.datamodel.dtos.business.services.PrimaryAnalysis
+import life.qbic.datamodel.dtos.business.services.Product
+import life.qbic.datamodel.dtos.business.services.ProjectManagement
+import life.qbic.datamodel.dtos.business.services.ProteomicAnalysis
+import life.qbic.datamodel.dtos.business.services.SecondaryAnalysis
+import life.qbic.datamodel.dtos.business.services.Sequencing
+
+import java.text.DateFormat
+
+/**
+ * <h1>Creates the content for an offer export</h1>
+ *
+ * <p>This use case aggregates all information that needs to be present in an offer. This use case should be called to forward the offer information
+ * </p>
+ *
+ * @since 1.1.0
+ *
+*/
+class CreateOfferContent implements CreateOfferContentInput, FetchOfferOutput{
+
+    CreateOfferContentOutput output
+    FetchOfferInput fetchOfferInput
+
+    private AffiliationCategory affiliationCategory
+    private double overheadRatio
+    private static List<Class> DATA_GENERATION = [Sequencing, ProteomicAnalysis, MetabolomicAnalysis]
+    private static List<Class> DATA_ANALYSIS = [PrimaryAnalysis, SecondaryAnalysis]
+    private static List<Class> PROJECT_AND_DATA_MANAGEMENT = [ProjectManagement, DataStorage]
+
+
+    private List<ProductItem> dataGenerationItems
+    private List<ProductItem> dataAnalysisItems
+    private List<ProductItem> dataManagementItems
+
+
+    CreateOfferContent(CreateOfferContentOutput output, FetchOfferDataSource fetchOfferDataSource){
+        this.output = output
+        this.fetchOfferInput = new FetchOffer(fetchOfferDataSource,this)
+    }
+
+    @Override
+    void createOfferContent(OfferId offerId) {
+        fetchOfferInput.fetchOffer(offerId)
+    }
+
+    @Override
+    void failNotification(String notification) {
+        output.failNotification(notification)
+    }
+
+    @Override
+    void fetchedOffer(Offer fetchedOffer) {
+        //create the offer business object
+        life.qbic.business.offers.Offer offer = Converter.convertDTOToOffer(fetchedOffer)
+        affiliationCategory = offer.selectedCustomerAffiliation.category
+        overheadRatio = offer.getOverheadRatio()
+
+        //collect the content for the offerpdf
+        DateFormat dateFormat = DateFormat.getDateInstance(DateFormat.LONG, Locale.US)
+        String creationDate = dateFormat.format(offer.modificationDate)
+        String expirationDate = offer.expirationDate.toLocalDate().toString()
+        groupProductItems(offer.items)
+
+        def offerContent = new OfferContent.Builder(offer.customer,offer.selectedCustomerAffiliation,offer.projectManager,creationDate,expirationDate,offer.projectTitle
+        ,offer.projectObjective, offer.experimentalDesign.orElse(""),offer.identifier.toString())
```

Comment:
Here, your `offerContent` is instanceof `OfferContent.Builder` currently. So I would suggest to rename it. That removes the risk of confusing the builder with the actual instance and only using the builder in the code that follows.
```suggestion
        def offerContentBuilder = new OfferContent.Builder(offer.customer,offer.selectedCustomerAffiliation,offer.projectManager,creationDate,expirationDate,offer.projectTitle
        ,offer.projectObjective, offer.experimentalDesign.orElse(""),offer.identifier.toString())
```

Reply:
I like your suggestion